packages/xod-client/src/editor/containers/Patch/modes/selecting.jsx

```diff
@@ -39,25 +40,31 @@ const selectingMode = {
     if (isEntitySelected(entityType, props.selection, entityId)) {
       if (isSelectionModifierPressed(event)) {
         props.actions.deselectEntity(entityType, entityId);
-        setState({ isMouseDownOnMovableObject: false });
+        setState({
+          isMouseDown: false,
+          isMouseDownOnMovableObject: false,
+        });
       } else {
         // Don't set selection to this single entity
         // to allow user to move already selected group of entities.
         // We'll handle it in `onEntityMouseUp`
         setState({
+          isMouseDown: true,
           isMouseDownOnMovableObject: true,
           dragStartPosition: mousePosition,
         });
       }
     } else if (isSelectionModifierPressed(event)) {
       props.actions.addEntityToSelection(entityType, entityId);
       setState({
+        isMouseDown: true,
         isMouseDownOnMovableObject: true,
         dragStartPosition: mousePosition,
       });
     } else {
       props.actions.selectEntity(entityType, entityId);
       setState({
+        isMouseDown: true,
```

Comment:
I don't think all those `isMouseDown`s are necessary. Maybe, a single `isMouseDownOnBackground` is clearer?